sycl/include/sycl/ext/oneapi/experimental/cluster_group_prop.hpp

```diff
@@ -0,0 +1,78 @@
+//==--- cluster_group_prop.hpp --- SYCL extension for cuda cluster group ---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <sycl/ext/oneapi/properties/properties.hpp>
+#include <sycl/range.hpp>
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi::experimental {
+
+namespace cuda {
+template <int Dim>
+struct cluster_size
+    : ::sycl::ext::oneapi::experimental::detail::run_time_property_key<
+          ::sycl::ext::oneapi::experimental::detail::ClusterLaunch> {
+  cluster_size(const range<Dim> &size) : size(size) {}
+  sycl::range<Dim> get_cluster_size() { return size; }
+  range<Dim> size;
```

Comment:
This member should maybe be private.

Reply:
Maybe, but I'm not sure. I doubt that this is very important either way but I didn't design the `cluster_size` property, so I would prefer not to touch it at this stage, especially since the CI is brittle and we have finally gotten all green and no conflicts. This is an experimental interface and minor things like this can be easily changed later.
The important thing is that we get this feature in some form merged asap. Apparently this should be available in a nightly build asap @Ruyk ?

Soon I will be on holiday and will be unable to continue refactoring it as new abi-break PRs and unified runtime changes are merge, such that it will be at risk of missing the abi-break window at all.

Reply:
@intel/dpcpp-tools-reviewers 

This is 100% green now. Please could you review this asap.

Reply:
@AerialMantis Made the member `size` private in the [latest commit](https://github.com/intel/llvm/pull/14113/commits/91bc7afeccc3c40a065b451a1a0f26c37946e1c1)